### PR TITLE
You can now help intent click on people to complete a dap, highfive, or handshake

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

You can now help intent click on people to complete a dap, highfive, or handshake. Also does some refactoring of the nearby code

## Why It's Good For The Game
Makes it easier to do stuff like handshakes without binds

## Testing

Tested it, realized I put in some wrong code, tested it again and it worked

## Changelog
:cl:
add: You can now help intent click on people to complete a dap, highfive, or handshake
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -219,53 +219,58 @@
...
-							H.wear_suit.add_fingerprint(M)
-						else if(H.w_uniform)
-							H.w_uniform.add_fingerprint(M)
+	if(health < HEALTH_THRESHOLD_CRIT)
+		return
+	if(src == M && ishuman(src))
+		check_self_for_injuries()
+		return
+	if(player_logged)
+		M.visible_message("<span class='notice'>[M] shakes [src], but [p_they()] [p_do()] not respond. Probably suffering from SSD.", \
```
Comment (lewcc) [inline code/modules/mob/living/carbon/carbon.dm:228]:
```suggestion
		M.visible_message("<span class='notice'>[M] shakes [src], but [p_they()] [p_do()] not respond. Probably suffering from SSD.</span>", \
```

File: code/datums/status_effects/status_effect.dm [hunk 2]
```diff
@@ -146,6 +146,15 @@
 			if(initial(S1.id) == S.id)
 				return S
 
+/// Returns the effect if the mob calling the proc owns the given status effect, but checks by type.
+/mob/living/proc/has_status_effect_type(effect)
+	. = null
+	if(!status_effects)
```
Comment (lewcc) [inline code/datums/status_effects/status_effect.dm:152]:
```suggestion
	if(!length(status_effects))
```

Comment (lewcc) [inline code/datums/status_effects/status_effect.dm:152]:
```suggestion
	if(!length(status_effects))
```

File: code/datums/status_effects/status_effect.dm [hunk 3]
```diff
@@ -146,6 +146,15 @@
 			if(initial(S1.id) == S.id)
 				return S
 
+/// Returns the effect if the mob calling the proc owns the given status effect, but checks by type.
+/mob/living/proc/has_status_effect_type(effect)
+	. = null
```
[the remark this answers is not in the document]
Comment (Contrabang) [reply code/datums/status_effects/status_effect.dm:151]: I was just doing it to be consistent with the other procs, I'll remove it

File: code/datums/status_effects/status_effect.dm (hunk 2 above)
Comment (Contrabang) [reply code/datums/status_effects/status_effect.dm:152]: Fixed

**Review by S34NW — APPROVED** [review]

**Pull request merged**